aiida/tools/graph/age_entities.py

```diff
@@ -0,0 +1,487 @@
+            inputs_len = len(input_for_set)
+            inside_len = len(self._edge_identifiers)
+            error_message = 'tuple passed has len = {}, but there are {} identifiers'
+            raise ValueError(error_message.format(inputs_len, inside_len))
+        return input_for_set
+
+    def get_template(self):
+        return DirectedEdgeSet(aiida_cls_to=self.aiida_cls_to, aiida_cls_from=self.aiida_cls_from)
+
+    @property
+    def aiida_cls_to(self):
+        """The class of nodes which the edge points to"""
+        return self._aiida_cls_to
+
+    @property
+    def aiida_cls_from(self):
+        """The class of nodes which the edge points from"""
+        return self._aiida_cls_from
+
+    @property
+    def edge_namedtuple(self):
+        """The namedtuple type used for the edges` identifiers"""
+        return self._edge_namedtuple
+
+    @property
+    def edge_identifiers(self):
+        """The identifiers for the edges"""
+        return self._edge_identifiers
+
+
+class Basket():
+    """Container for several instances of
+    :py:class:`aiida.tools.graph.age_entities.AiidaEntitySet` .
+
+    In the current implementation, it contains one EntitySet for Nodes and one for Groups,
+    and one EdgeSet for Node-Node edges (links) and one for Group-Node connections.
+    """
+
+    def __init__(self, nodes=None, groups=None, nodes_nodes=None, groups_nodes=None):
+        """Initialization method
+
+        During initialization of the basket, both the sets of nodes and the set of
+        groups can be provided as one of the following: an AiidaEntitySet with the
+        respective type (node or group) or a list/set/tuple with the ids of the
+        respective node or group.
+
+        :param nodes: AiiDA nodes provided in an acceptable way.
+        :param groups: AiiDA groups provided in an acceptable way.
+        """
+
+        def get_check_set_entity_set(input_object, keyword, aiida_class):
+
+            if input_object is None:
+                output_set = AiidaEntitySet(aiida_class)
+                return output_set
+
+            if isinstance(input_object, (list, tuple, set)):
+                output_set = AiidaEntitySet(aiida_class)
+                output_set.set_entities(input_object)
+                return output_set
+
+            if isinstance(input_object, AiidaEntitySet):
+                if input_object.aiida_cls is aiida_class:
+                    return input_object
+                raise TypeError('{}  has to  have {} as aiida_cls'.format(keyword, aiida_class))
+
+            else:
+                error_message = 'Input object is of type {}.\n'.format(type(input_object))
+                error_message += 'Instead, it should be either None or one of:\n'
+                error_message += ' - {}\n'.format(AiidaEntitySet)
+                error_message += ' - {}\n'.format(list)
+                error_message += ' - {}\n'.format(tuple)
+                error_message += ' - {}\n'.format(set)
```

Comment:
Bit excessive to use three lines where you can just say "should be list, tuple or set"